internal/corazawaf/body_buffer.go

```diff
@@ -45,13 +45,22 @@ func (br *BodyBuffer) WriteTo(w io.Writer) (int64, error) {
 
 // Write appends data to the body buffer by chunks
 // You may dump io.Readers using io.Copy(br, reader)
+// Note: remember to check overflow before calling Write
```

Comment:
Curious about this? Will upstream code need to do this check?

Reply:
this PR was originally adding the overflow check inside the `Write` function. With the latest commits, I moved it to `WriteRequestBody`/`ReadRequestBodyFrom` functions because `tx.requestBodyBuffer.length+writingBytes` is already performed there and an overflow would make not true the `>= tx.RequestBodyLimit` condition.

Moving this code out of the `Write` function, means that the latter is back on computing just `targetLen := br.length + int64(len(data)), this is the reason why I felt it was okay to add at least a note about checking overflow before calling it.

What do you think, would it be better, in any case, to still do this check also here, even if it would be double-checking?

Reply:
I would say you want to do the overflow check here as well. I think such double checking is fine.